core/src/test/scala/unit/kafka/server/ConsumerGroupDescribeRequestsTest.scala

```diff
@@ -0,0 +1,175 @@
+
+import scala.jdk.CollectionConverters._
+
+@Timeout(120)
+@ExtendWith(value = Array(classOf[ClusterTestExtensions]))
+@ClusterTestDefaults(clusterType = Type.KRAFT, brokers = 1)
+@Tag("integration")
+class ConsumerGroupDescribeRequestsTest(cluster: ClusterInstance) extends GroupCoordinatorBaseRequestTest(cluster) {
+
+  @ClusterTest(clusterType = Type.KRAFT, serverProperties = Array(
+    new ClusterConfigProperty(key = "group.coordinator.rebalance.protocols", value = "classic,consumer"),
+    new ClusterConfigProperty(key = "group.coordinator.new.enable", value = "true"),
+    new ClusterConfigProperty(key = "offsets.topic.num.partitions", value = "1"),
+    new ClusterConfigProperty(key = "offsets.topic.replication.factor", value = "1")
+  ))
+  def testConsumerGroupDescribeWithNewGroupCoordinator(): Unit = {
+    testConsumerGroupDescribe()
+  }
+
+  private def testConsumerGroupDescribe(): Unit = {
+    // Creates the __consumer_offsets topics because it won't be created automatically
+    // in this test because it does not use FindCoordinator API.
+    createOffsetsTopic()
+
+    val admin = cluster.createAdminClient()
+    val topicId = TestUtils.createTopicWithAdminRaw(
+      admin = admin,
+      topic = "foo",
+      numPartitions = 3
+    )
+
+    val timeoutMs = 5 * 60 * 1000
+    val clientId = "client-id"
+    val clientHost = "/127.0.0.1"
+
+    // Add first group
+    val consumerGroupHeartbeatResponseData1 = consumerGroupHeartbeat(
+      groupId = "grp-1",
+      rebalanceTimeoutMs = timeoutMs,
+      subscribedTopicNames = List(""),
+      topicPartitions = List.empty
+    )
+
+    // Add second group
+    consumerGroupHeartbeat(
+      memberId = "member-1",
+      groupId = "grp-2",
+      rebalanceTimeoutMs = timeoutMs,
+      subscribedTopicNames = List("foo"),
+      topicPartitions = List.empty
+    )
+
+    // Add second member to second group
+    consumerGroupHeartbeat(
+      memberId = "member-2",
+      groupId = "grp-2",
+      rebalanceTimeoutMs = timeoutMs,
+      subscribedTopicNames = List("foo"),
+      topicPartitions = List.empty
+    )
+
+    val assignmentMember1 = new Assignment()
+      .setTopicPartitions(List(
+        new TopicPartitions()
+          .setTopicId(topicId)
+          .setTopicName("foo")
+          .setPartitions(List[Integer](0, 1, 2).asJava)
+      ).asJava)
+
+    val targetAssignmentMember1 = new Assignment()
+      .setTopicPartitions(List(
+        new TopicPartitions()
+          .setTopicId(topicId)
+          .setTopicName("foo")
+          .setPartitions(List[Integer](0, 1).asJava)
+      ).asJava)
+
+    val assignmentMember2 = new Assignment()
+      .setTopicPartitions(List().asJava)
+
+    val targetAssignmentMember2 = new Assignment()
+      .setTopicPartitions(List(
+        new TopicPartitions()
+          .setTopicId(topicId)
+          .setTopicName("foo")
+          .setPartitions(List[Integer](2).asJava)
+      ).asJava)
```

Comment:
Would it make sense to inline those? I think that it is easier to read them when they are within their context.